Schedule update and tests

danny-avila opened the pull request.

danny-avila committed:
test(ScheduleView.spec.tsx): add tests for ScheduleView component and its child components

danny-avila committed:
test: add tests for ClassCard and ScheduleView components
feat: replace useSession with getServerSession in schedule page
fix: change authOptions import in schedule page
chore: update jest configuration to include coverage reporting
chore: add jest setup file to mock Swiper CSS and import testing-library dependencies
chore: update dependencies in package.json

refactor(ScheduleView.tsx): remove unused imports and functions
feat(ScheduleView.tsx): add getWeekTitle, renderWeekSlides, getScheduledDates, and getNextScheduledClass functions from date-fns.ts
test(date-fns.spec.tsx): add tests for getWeekTitle, renderWeekSlides, getScheduledDates, and getNextScheduledClass functions from date-fns.ts

feat(date-fns.tsx): add utility functions for parsing and formatting dates, rendering week days and slides, and getting week title.

danny-avila committed:
chore: update dependencies and add DateFunctions tests

fix: remove duplicate @testing-library/react dependency

feat: add DateFunctions module with functions for parsing dates, getting scheduled dates, getting next scheduled class, rendering week days, rendering week slides, and getting week title

refactor: sort user classes by date in Schedule component

refactor: use DateFunctions module in ScheduleView component

refactor: use addDays function to set active day in ScheduleView component

chore: remove unused exports from lib/index.ts

danny-avila committed:
fix(DateFunctions.tsx): add missing import statement for React
fix(ScheduleView.tsx): remove unused useCallback hook
feat(ScheduleView.tsx): add initialSlide, totalSlides, and today constants
feat(ScheduleView.tsx): refactor gotoClassDate function to use initialSlide and today constants
feat(ScheduleView.tsx): update initialSlide prop in Swiper component to use initialSlide constant
feat(ScheduleView.tsx): update renderWeekSlides function to use totalSlides constant

danny-avila committed:
chore(.eslintrc.json): add rules for devDependencies and optionalDependencies
feat(ScheduleView.tsx): add button disable functionality when clicking 'View All' button

danny-avila committed:
fix(DateFunctions.tsx): fix export statement for WeekDays component
feat(DateFunctions.tsx): add WeekDays component to render week days in ScheduleView component
fix(ScheduleView.tsx): fix import statements for Swiper and SwiperSlide components
feat(ScheduleView.tsx): add Virtual and Navigation modules to Swiper component
feat(ScheduleView.tsx): add navigation buttons to switch between weeks in Swiper component
feat(ScheduleView.tsx): add state and logic to track active slide and active day in ScheduleView component
feat(ScheduleView.tsx): add WeekDays component to render week days in Swiper component

danny-avila committed:
fix: scheduleview carousel bug, adjusts tests based on changes

danny-avila committed:
chore(eslint): comment out "test.{ts,tsx}" in devDependencies rule
feat(eslint): add override rule for "test.{ts,tsx}" files to set devDependencies to 2

danny-avila committed:
chore(.eslintrc.json): remove unnecessary devDependencies array
chore(.eslintrc.json): change optionalDependencies value to 0
chore(Schedule.tsx): change console.log message to 'schedule data'
chore(ScheduleView.spec.tsx): format import statements

chore(ScheduleView.tsx): organize imports and format code
feat(ScheduleView.tsx): add key prop to SwiperSlide component
feat(ScheduleView.tsx): add format function from date-fns library
feat(ScheduleView.tsx): add key prop to SwiperSlide component in map function

danny-avila committed:
refactor: server-side rendering for schedule classes

kahlstorf1 commented on src/components/Schedule/ScheduleView.tsx:
```diff
@@ -132,38 +67,60 @@ export default function ScheduleView({ userClasses }: ScheduleViewProps) {
 
         </button>
       </div>
+      <p>{`State is ${activeSlide}`}</p>
```
Might want to remove this if functionality is fully built

kahlstorf1 reviewed (changes requested):
Looks good to me, Jest suite looks good as well. Feel free to merge once state is removed

danny-avila committed:
chore(schedule/page.tsx): update console.log statement to include userId
chore(Nav.tsx): comment out welcome message for session user
chore(ScheduleView.tsx): update initialSlide and totalSlides values

danny-avila committed:
chore(package.json): add start-p script to start the server on port 3002

danny-avila committed:
chore(ScheduleView.tsx): remove unnecessary state display

kahlstorf1 reviewed (approved):
Looks good!
